src/PageUtils/Devices/DeviceFactory.php

```diff
@@ -0,0 +1,585 @@
+<?php
+
+namespace HeadlessChromium\PageUtils\Devices;
+
+use HeadlessChromium\Exception\InexistentDevice;
+use HeadlessChromium\PageUtils\Devices\Device;
+use HeadlessChromium\PageUtils\Devices\Resolution;
+use HeadlessChromium\PageUtils\Devices\Viewport;
+
+final class DeviceFactory
+{
+    /**
+     * @var array<string, Device> $devices
+     */
+    private array $devices = [];
+
+    /** 
+     * @return void
+     */
+    public function __construct()
+    {
+        foreach ($this->devices() as $device) {
+            $deviceViewport = $device['viewport'];
+
+            $resolution = new class($deviceViewport['width'], $deviceViewport['height'], $deviceViewport['deviceScaleFactor']) extends Resolution {};
+        
+            $viewport = new class($resolution, $deviceViewport['isMobile'], $deviceViewport['hasTouch'], $deviceViewport['isLandscape']) extends Viewport {};
+        
+            $device = new class($device['name'], $device['user-agent'], $viewport) extends Device {};
```

Comment:
Need to be fixed because of variable `$device` collision

Reply:
I don't see a reason for those 3 different classes. They seem to be just data classes with fairly few values. Why not have just a single class?

Reply:
> 

I can agree with something at the point of nonsense with these classes in the general purpose of this package. But with SOLID in mind and perspective of reusing the abstraction I had to prefer these classes. I will look at this point later again.

Reply:
And yes, there is a transitional dependency with Device::class and Resolution::class through Viewport::class which is not good for Low Coupling.